datafusion/core/src/execution/context/mod.rs

```diff
@@ -1790,6 +1837,38 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_dynamic_file_query() -> Result<()> {
+        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let path = path.join("tests/tpch-csv/customer.csv");
+        let url = format!("file://{}", path.display());
+        let cfg = SessionConfig::new().set_str("datafusion.catalog.has_header", "true");
+        let session_state = SessionStateBuilder::new()
+            .with_default_features()
+            .with_config(cfg)
+            .build();
+        let ctx = SessionContext::new_with_state(session_state).enable_url_table();
+        let result = plan_and_collect(
+            &ctx,
+            format!("select c_name from '{}' limit 3;", &url).as_str(),
+        )
+        .await?;
```

Comment:
I wonder if it would be possible here to add `enable_url_table` as a argument to SessionStateBuilder

So it could look like

```rust
let session_state =  SessionStateBuilder::new()
            .with_default_features()
            .with_config(cfg)
            .enable_url_table()
            .build();
let ctx = SessionContext::from(session_state);
```

🤔  

(we can do this as a follow on as well)

Reply:
Sure. I think it's a good idea. We can have a flag in the builder. If it's true, we can wrap the catalog list implicitly. It may need some additional implementation and test cases. I prefer to do it in the follow-up PR.